[quantin96] iP

quantin96 opened the pull request:
# Dorito
> "The key is to have fun, even if you suck!"

Introducing **Dorito** ⛛, a cute chatbot that keeps track of your everyday tasks! It is
- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

Start using **Dorito** with the following steps:
1. Download **Dorito** from *[here](https://github.com/quantin96/ip)*
2. Ddd your tasks
3. Let it manage your tasks for you! 😉

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
#
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Dorito {
    public static void main(String[] args) throws IOException {
        new Dorito("./data/dorito.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

quantin96 committed:
Renamed chatbot and implemented greet and exit

quantin96 committed:
Implemented echo feature

quantin96 committed:
Implemented add tasks to list feature

quantin96 committed:
Added Task class and implemented a checkbox for tasks

quantin96 committed:
Implemented sub-classes of Task (ToDo, Deadline, Event) and updated output format

quantin96 committed:
Updated Automated Text UI Testing for Dorito

quantin96 committed:
Created DoritoException and implemented error-handling for two cases (invalid ToDo tasks and invalid commands)

quantin96 committed:
Changed implementation of tasks from Array to ArrayList and implemented Delete function

quantin96 committed:
Add updateFile and loadFile

quantin96 committed:
Merge branch 'branch-Level-7'

quantin96 committed:
Add date; Modify Deadline.java, Event.java

quantin96 committed:
Merge branch 'branch-Level-8'

quantin96 committed:
Add Storage and Ui

quantin96 committed:
Merge branch 'branch-OOP'

quantin96 committed:
Add dorito package

quantin96 committed:
Merge branch 'branch-package'

quantin96 committed:
Merge branch 'master' into add-gradle-support

quantin96 committed:
Add Gradle

quantin96 committed:
Merge branch 'add-gradle-support'

quantin96 committed:
Add JavaDoc

quantin96 committed:
Fix coding standard

quantin96 committed:
Add find method

quantin96 committed:
Merge branch 'branch-A-JavaDoc'

quantin96 committed:
Merge branch 'branch-A-CodingStandard'

quantin96 committed:
Merge branch 'branch-Level-9'

quantin96 committed:
Add Parser.java

quantin96 committed:
Add GUI

quantin96 committed:
Merge branch 'branch-Level-10'

yuqiannemo commented on src/main/java/dorito/Parser.java:
```diff
@@ -0,0 +1,92 @@
+    public enum Commands {
+        BYE, LIST, FIND, TODO, DEADLINE, EVENT, MARK, UNMARK, DELETE
+    }
+
```
Nice naming for command enums here!

yuqiannemo commented on src/main/java/dorito/Parser.java:
```diff
@@ -0,0 +1,92 @@
+            case DEADLINE:
+                String deadlineSub = input.substring(9);
+                String deadlineDesc = deadlineSub.split(" /by ")[0];
+                String by = deadlineSub.split(" /by ")[1];
```
The naming of "task type + Sub" and "task type + Desc" (e.g. deadlineSub, deadlineDesc) is consistent across the deadline and event commands. However, would it be better to replace them with "message" and "description", which does not use abbreviations and might be more straightforward?

yuqiannemo commented on src/main/java/dorito/Parser.java:
```diff
@@ -0,0 +1,92 @@
+                break;
+            case UNMARK:
+                Task unmark = tasks.get(Integer.parseInt(input.split(" ")[1]) - 1);
+                unmark.unmark();
```
Would it be better to change `Task unmark = tasks.get(Integer.parseInt(input.split(" ")[1]) - 1);`
to `Task taskToUnmark = tasks.get(Integer.parseInt(input.split(" ")[1]) - 1);` to make it clearer?

yuqiannemo commented on src/main/java/dorito/Storage.java:
```diff
@@ -0,0 +1,81 @@
+        }
+        Scanner sc = new Scanner(f);
+        DateTimeFormatter dateInput = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        DateTimeFormatter dateOutput = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```
The naming here LGTM and indicates a good coding quality!

yuqiannemo commented on src/main/java/dorito/Task.java:
```diff
@@ -0,0 +1,33 @@
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Clear and straightforward naming!

yuqiannemo reviewed (commented):
Review for code quality (naming)

Xzh119 commented on src/main/java/dorito/MainWindow.java:
```diff
@@ -0,0 +1,61 @@
+ * Controller for the main GUI.
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
+
+    private Dorito dorito;
```
Good use of a camelCase naming

Xzh119 commented on src/main/java/dorito/Parser.java:
```diff
@@ -0,0 +1,92 @@
+        try {
+            Commands command = Commands.valueOf(input.split(" ")[0].toUpperCase());
+            switch(command) {
+            case BYE:
+                output = ui.bye();
+                break;
+            case LIST:
+                output = ui.list(tasks);
+                break;
+            case FIND:
+                output = ui.find(tasks, input.split(" ")[1]);
+                break;
+            case TODO:
+                try {
```
The indentation for case clauses is great.

Xzh119 commented on src/main/java/dorito/MainWindow.java:
```diff
@@ -0,0 +1,61 @@
+    public void setDorito(Dorito d) {
+        dorito = d;
+
+        dialogContainer.getChildren().addAll(DialogBox.getDoritoDialog(d.start(), doritoImage));
```
Maybe the code can add some helper variables to make it more neat and easy to understand.

Xzh119 commented on src/main/java/dorito/DoritoException.java:
```diff
@@ -0,0 +1,8 @@
+package dorito;
+
+public class DoritoException extends Exception {
+
+    public DoritoException (String message){
+        super(message);
+    }
```
Maybe can have more exception situations?

Xzh119 commented on src/main/java/dorito/Ui.java:
```diff
@@ -0,0 +1,126 @@
+    /**
+     * Checks if the program has exited.
+     */
+    public boolean isExit() {
+        return exit;
+    }
```
Good use of boolean naming

Xzh119 commented on src/main/java/dorito/Ui.java:
```diff
@@ -0,0 +1,126 @@
+     *
+     * @param task The task to be added.
+     * @param i Number of tasks in task list.
+     */
```
Good and clear comments

Xzh119 reviewed (commented):
Overall, I found your code easy to read for the most part except a few places
where the nesting was too deep. I noted a few minor coding standard violations
too. Some of the classes are getting quite long. Consider splitting into
smaller classes if that makes sense.

quantin96 committed:
Add assertions

Added assertions to verify Dorito runs as expected

quantin96 committed:
Fix code quality issues

Redelegated tasks from long methods to extract methods such that the code looks cleaner.

quantin96 committed:
Merge pull request #1 from quantin96/branch-A-Assertions

Add assertions

quantin96 committed:
Merge branch 'master' into branch-A-CodeQuality

quantin96 committed:
Merge pull request #2 from quantin96/branch-A-CodeQuality

Fix code quality issues

quantin96 committed:
Add priority

Implemented task priority levels HIGH, MEDIUM, LOW, DEFAULT.
Users can now set a priority for tasks.

Features to be added:
* filter by priority level.
* sort tasks by priority level.

quantin96 committed:
Merge branch 'branch-C-Priority'

quantin96 committed:
Refactor code and add more error handling and tests

quantin96 committed:
Refactor error messages

quantin96 committed:
Add Ui.png

quantin96 committed:
Update User Guide

quantin96 committed:
Update Storage.java

quantin96 committed:
Fix Event bug
